lms/djangoapps/labster_accounts/management/commands/obfuscate_user_data.py

```diff
@@ -0,0 +1,154 @@
+        )
+
+    def handle(self, *args, **options):
+        """
+        Execute the command.
+        """
+        licenses = options['licenses']
+        emails = options['emails']
+
+        user_emails = set()
+
+        if licenses:
+            for lic in licenses:
+                print ("Processing license %s" % lic)
+                try:
+                    ccx = CourseLicense.objects.get(license_code=lic).course_id
+
+                    enrolled_users = CourseEnrollment.objects.filter(course_id=ccx).values_list(
+                        'user__email', flat=True
+                    ).exclude(user__email__contains='labster.com')
+                    not_enrolled_users = CourseEnrollmentAllowed.may_enroll_and_unenrolled(ccx).values_list(
+                        'email', flat=True
+                    )
+
+                    all_users = chain(not_enrolled_users, enrolled_users)
+
+                    user_emails |= set(all_users)
+                except CourseLicense.DoesNotExist:
+                    pass
+
+        if emails:
+            user_emails |= set(emails)
+
+        self.obfuscate_users(user_emails)
+
+    def obfuscate_users(self, user_emails):
+        """
+        Obfuscates user model fields: names, email.
+        """
+        users = User.objects.filter(email__in=user_emails)
+
+        field_actions = {
+            'first_name': self.generate_random_string,
+            'last_name': self.generate_random_string,
+            'username': self.generate_random_string,
+            'profile.name': self.generate_random_string,
+            'profile.mailing_address': self.generate_random_string,
+            'email': self.obfuscate_email
+        }
+
+        self.obfuscate_names_emails(users, field_actions)
+
+    @staticmethod
+    def obfuscate_names_emails(query_set, field_actions):
+        """
+        Obfuscates objects name, email and other fields.
+        """
+        fields_select = set([key.split('.')[0] if '.' in key else key for key in field_actions.keys()])
+        for obj in query_set.only(*fields_select):
+            print("Obfuscating data for %s" % obj)
+            values = {}
+            references = {}
+            for field, action in field_actions.items():
+                action = action_decorator(action, field)
+                if '.' in field:
+                    # one2one relation or foreign key found
+                    names = field.split('.')
+                    # get relation field value
+                    val = getattr(getattr(obj, names[0]), names[1], '')
+                    if not (names[0] in references):
+                        references[names[0]] = {}
+                    references[names[0]].update({names[1]: action(val)})
+                else:
+                    val = getattr(obj, field, '')
+                    values[field] = action(val)
+
+            if values:
+                obj.__class__.objects.filter(id=obj.id).update(**values)
```

Comment:
Should we handle Integrity error here? If obfuscated email duplicates for some reasons?

Reply:
Do we need transaction.atomic per student?

Reply:
added both